src/post.ts

```diff
@@ -94,16 +94,39 @@ async function reportAll(
   }
 
   const commentOnPR: string = core.getInput('comment_on_pr')
-  if (pull_request && 'true' === commentOnPR) {
+  if (pull_request && ['true', 'update'].indexOf(commentOnPR) >= 0) {
     if (logger.isDebugEnabled()) {
       logger.debug(`Found Pull Request: ${JSON.stringify(pull_request)}`)
     }
 
-    await octokit.rest.issues.createComment({
-      ...github.context.repo,
-      issue_number: Number(github.context.payload.pull_request?.number),
-      body: postContent
-    })
+    const issueNumber = Number(github.context.payload.pull_request?.number);
+    let createComment: boolean = true
+    if ('update' === commentOnPR) {
+      const comments = await octokit.rest.issues.listComments({
```

Comment:
Hi @bojanrajh, 
Thanks for your contribution. 
Sorry for late reply.
Can you also handle the case where there are more comments than the page size so there are multiple pages of issue comments? You may need to iterate over all pages to be sure that you checked all comments.

Reply:
@serkan-ozal Rebased and wrapped `listComments()` into a do-while loop with `page` parameter.